x-pack/plugins/transform/common/api_schemas/transforms.ts

```diff
@@ -86,18 +127,30 @@ export interface PutTransformsResponseSchema {
 }
 
 // POST transforms/_preview
-export const postTransformsPreviewRequestSchema = schema.object({
-  pivot: pivotSchema,
-  source: sourceSchema,
-});
+export const postTransformsPreviewRequestSchema = schema.object(
+  {
+    pivot: schema.maybe(pivotSchema),
+    latest: schema.maybe(latestFunctionSchema),
+    source: sourceSchema,
+  },
+  {
+    validate: transformConfigPayloadValidator,
+  }
+);
+
+export type PostTransformsPreviewRequestSchema = TypeOf<typeof postTransformsPreviewRequestSchema>;
 
-export interface PostTransformsPreviewRequestSchema
-  extends TypeOf<typeof postTransformsPreviewRequestSchema> {
+export type PivotTransformPreviewRequestSchema = Omit<
+  PostTransformsPreviewRequestSchema,
+  'latest'
+> & {
   pivot: {
     group_by: PivotGroupByDict;
     aggregations: PivotAggDict;
   };
-}
+};
+
+export type LatestTransformPreviewRequestSchema = Omit<PostTransformsPreviewRequestSchema, 'pivot'>;
```

Comment:
Great work with the types to work around the schema limitations with the custom validator!